core/src/main/java/org/apache/calcite/sql/fun/SqlArgumentAssignmentOperator.java

```diff
@@ -47,4 +47,8 @@
     writer.keyword(getName());
     call.operand(0).unparse(writer, getRightPrec(), rightPrec);
   }
+
+  @Override public boolean argumentMustBeScalar(int ordinal) {
+    return false;
```

Comment:
This is really nice. I actually tried to add named parameter support but failed on make non-scalar work.

I wasn't aware this can be as simple as override this parameter in SqlArgumentAssignmentOperator.java